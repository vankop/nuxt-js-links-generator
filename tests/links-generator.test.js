const path = require("path");
const { generateESModule, generateCommonJSModule } = require("../lib/util");
const generate = require("../lib/links-generator");

const fixtures = path.resolve(__dirname, "fixtures");

function fixture(f) {
    return path.resolve(fixtures, f);
}

it("app1. no pages", async (done) => {
    const links = await generate(fixture("app1"), false);
    expect(links).toMatchSnapshot();
    expect(generateESModule(links)).toMatchSnapshot();
    expect(generateCommonJSModule(links)).toMatchSnapshot();
    done();
});

it("app2", async (done) => {
    const links = await generate(fixture("app2"), false);
    expect(links).toMatchSnapshot();
    expect(generateESModule(links)).toMatchSnapshot();
    expect(generateCommonJSModule(links)).toMatchSnapshot();
    done();
});

it("app3", async (done) => {
    const links = await generate(fixture("app3"), false);
    expect(links).toMatchSnapshot();
    expect(generateESModule(links)).toMatchSnapshot();
    expect(generateCommonJSModule(links)).toMatchSnapshot();
    done();
});
