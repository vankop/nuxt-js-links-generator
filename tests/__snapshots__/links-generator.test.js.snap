// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app1. no pages 1`] = `Array []`;

exports[`app1. no pages 2`] = `
"/* eslint-disable */
/* tslint:disable */
// Generated by nuxt-js-links-generator@0.2.1
"
`;

exports[`app1. no pages 3`] = `
"/* eslint-disable */
/* tslint:disable */
// Generated by nuxt-js-links-generator@0.2.1
\\"use strict\\";

module.exports = {}"
`;

exports[`app2 1`] = `
Array [
  "function servicesPage() { return \\"/services\\" }",
  "function servicesOptionsPage() { return \\"/services/options\\" }",
]
`;

exports[`app2 2`] = `
"/* eslint-disable */
/* tslint:disable */
// Generated by nuxt-js-links-generator@0.2.1
export function servicesPage() { return \\"/services\\" }
export function servicesOptionsPage() { return \\"/services/options\\" }"
`;

exports[`app2 3`] = `
"/* eslint-disable */
/* tslint:disable */
// Generated by nuxt-js-links-generator@0.2.1
\\"use strict\\";
function servicesPage() { return \\"/services\\" }
function servicesOptionsPage() { return \\"/services/options\\" }
module.exports = {servicesPage, servicesOptionsPage}"
`;

exports[`app3 1`] = `
Array [
  "function customerById(_id: any) { return \`/customer/\${_id}\` }",
  "function servicesById(_id: any) { return \`/services/\${_id}\` }",
  "function servicesErrorIdByTypeAndIndex(_type: any, _index: any) { return \`/services/\${_type}/error-id/\${_index}\` }",
  "function servicesPage() { return \\"/services\\" }",
]
`;

exports[`app3 2`] = `
"/* eslint-disable */
/* tslint:disable */
// Generated by nuxt-js-links-generator@0.2.1
export function customerById(_id: any) { return \`/customer/\${_id}\` }
export function servicesById(_id: any) { return \`/services/\${_id}\` }
export function servicesErrorIdByTypeAndIndex(_type: any, _index: any) { return \`/services/\${_type}/error-id/\${_index}\` }
export function servicesPage() { return \\"/services\\" }"
`;

exports[`app3 3`] = `
"/* eslint-disable */
/* tslint:disable */
// Generated by nuxt-js-links-generator@0.2.1
\\"use strict\\";
function customerById(_id: any) { return \`/customer/\${_id}\` }
function servicesById(_id: any) { return \`/services/\${_id}\` }
function servicesErrorIdByTypeAndIndex(_type: any, _index: any) { return \`/services/\${_type}/error-id/\${_index}\` }
function servicesPage() { return \\"/services\\" }
module.exports = {customerById, servicesById, servicesErrorIdByTypeAndIndex, servicesPage}"
`;
